experiments/AtmosLES/stable_bl_model.jl

```diff
@@ -0,0 +1,324 @@
+    return nothing
+end
+
+"""
+  StableBL Sponge (Source)
+"""
+struct StableBLSponge{FT} <: Source
+    "Maximum domain altitude (m)"
+    z_max::FT
+    "Altitude at with sponge starts (m)"
+    z_sponge::FT
+    "Sponge Strength 0 ⩽ α_max ⩽ 1"
+    α_max::FT
+    "Sponge exponent"
+    γ::FT
+    "Eastward geostrophic velocity `[m/s]` (Base)"
+    u_geostrophic::FT
+    "Eastward geostrophic velocity `[m/s]` (Slope)"
+    u_slope::FT
+    "Northward geostrophic velocity `[m/s]`"
+    v_geostrophic::FT
+end
+function atmos_source!(
+    s::StableBLSponge,
+    atmos::AtmosModel,
+    source::Vars,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+    direction,
+)
+
+    z_max = s.z_max
+    z_sponge = s.z_sponge
+    α_max = s.α_max
+    γ = s.γ
+    u_geostrophic = s.u_geostrophic
+    u_slope = s.u_slope
+    v_geostrophic = s.v_geostrophic
+
+    z = altitude(atmos, aux)
+    u_geo = SVector(u_geostrophic + u_slope * z, v_geostrophic, 0)
+    ẑ = vertical_unit_vector(atmos, aux)
+    # Accumulate sources
+    if z_sponge <= z
+        r = (z - z_sponge) / (z_max - z_sponge)
+        β_sponge = α_max * sinpi(r / 2)^s.γ
+        source.ρu -= β_sponge * (state.ρu .- state.ρ * u_geo)
+    end
+    return nothing
+end
+
+"""
+  Initial Condition for StableBoundaryLayer LES
+"""
+function init_problem!(problem, bl, state, aux, localgeo, t)
+    (x, y, z) = localgeo.coord
+    # Problem floating point precision
+    FT = eltype(state)
+    R_gas::FT = R_d(bl.param_set)
+    c_p::FT = cp_d(bl.param_set)
+    c_v::FT = cv_d(bl.param_set)
+    p0::FT = MSLP(bl.param_set)
+    _grav::FT = grav(bl.param_set)
+    γ::FT = c_p / c_v
+    # Initialise speeds [u = Eastward, v = Northward, w = Vertical]
+    u::FT = 8
+    v::FT = 0
+    w::FT = 0
+    # Assign piecewise quantities to θ_liq and q_tot
+    θ_liq::FT = 0
+    q_tot::FT = 0
+    # Piecewise functions for potential temperature and total moisture
+    z1 = FT(100)
+    if z <= z1
+        θ_liq = FT(265)
+    else
+        θ_liq = FT(265) + FT(0.01) * (z - z1)
+    end
+    θ = θ_liq
+    π_exner = FT(1) - _grav / (c_p * θ) * z # exner pressure
+    ρ = p0 / (R_gas * θ) * (π_exner)^(c_v / R_gas) # density
+    # Establish thermodynamic state and moist phase partitioning
+    TS = PhaseEquil_ρθq(bl.param_set, ρ, θ_liq, q_tot)
+
+    # Compute momentum contributions
+    ρu = ρ * u
+    ρv = ρ * v
+    ρw = ρ * w
+
+    # Compute energy contributions
+    e_kin = FT(1 // 2) * (u^2 + v^2 + w^2)
+    e_pot = _grav * z
+    ρe_tot = ρ * total_energy(e_kin, e_pot, TS)
+
+    # Assign initial conditions for prognostic state variables
+    state.ρ = ρ
+    state.ρu = SVector(ρu, ρv, ρw)
+    state.ρe = ρe_tot
+    state.moisture.ρq_tot = ρ * q_tot
+
+    if z <= FT(50) # Add random perturbations to bottom 50m of model
+        state.ρe += rand() * ρe_tot / 100
+    end
+end
+
+function surface_temperature_variation(state, aux, t)
```

Comment:
```suggestion
function surface_temperature_variation(state, t)
```
Better to drop `aux` here if we're not using it. See #1648.